### PR TITLE
Improvement: Improve identify confounders function

### PR DESCRIPTION
**Pull request by andrew-causalens**
## Description and Motivation

Fix some error cases, improve memory usage, and extend to `NodeLike`. No changelog necessary as the previous feature was not released. Therefore prior message is sufficient.

## How Has This Been Tested?

New unit tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change which improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactor (moving code around, general refactoring improvements)
- [ ] Documentation (documentation)

## PR Checklist:

- [x] I have reviewed my own PR? (review the PR as you are reviewing someone else's PR)
- [x] I have implemented all requirements? (see JIRA, project documentation)
- [x] I am affecting someone else's work? If so: I have added those people as reviewers?
- [ ] I have added comments to all the bits that are hard to follow
- [ ] At a bare minimum, I tested this manually
- [ ] I have added unit test(s)
- [ ] Is this a bugfix? If so have I added a regression test?
- [x] Does this PR satisfy the [one PR, one concern](https://medium.com/@fagnerbrack/one-pull-request-one-concern-e84a27dfe9f1) rule?
- [ ] If needed, documentation has been added/updated?
- [ ] I have updated the appropriate changelog with a line for my changes
- [ ] If this PR breaks reproducibility, have I added a note in the changelog explaining the break in reproducibility
      and its extent?

### CODE REVIEW

File: cai_causal_graph/identify_utils.py [hunk 1]
```diff
@@ -43,43 +46,76 @@ def identify_confounders(graph: CausalGraph, node_1: str, node_2: str) -> List[s
...
+        raise CausalGraphErrors.NodeDoesNotExistError(f'Node not found: {node_2}')
+
+    # Coerce NodeLike to identifier. We already know they are NodeLike as node_exists does this.
+    node_1_id = Node.identifier_from(node_1)
+    node_2_id = Node.identifier_from(node_2)
+
+    # Ensure node_1 != node_2
+    if node_1_id == node_2_id or node_1 == node_2:
+        raise ValueError('node_1 and node_2 cannot be equal. Please provide different nodes / node identifiers.')
+
```
Comment (steven-causalens) [inline cai_causal_graph/identify_utils.py:111]: I still believe these checks can be abstracted for future use, but I will just do that in an upcoming PR where I add `identify_instruments` and `identify_mediators`

**Review by steven-causalens — APPROVED** [review]
LGTM, thanks @andrew-causalens!

File: cai_causal_graph/causal_graph.py
[the remark this answers is not in the document]
Comment (andrew-causalens) [reply cai_causal_graph/causal_graph.py:1]: No. Discussed in person.

File: cai_causal_graph/identify_utils.py (hunk 1 above)
Comment (andrew-causalens) [reply cai_causal_graph/identify_utils.py:111]: Will be done on later PR that would reuse this code.

**Pull request merged**